src/main/fc/runtime_config.c

```diff
@@ -34,34 +34,37 @@ static uint32_t enabledSensors = 0;
 
 // Must be no longer than OSD_WARNINGS_MAX_SIZE (11) to be displayed fully in OSD
 const char *armingDisableFlagNames[]= {
-    "NOGYRO",
-    "FAILSAFE",
-    "RXLOSS",
-    "NOT_DISARMED",
-    "BOXFAILSAFE",
-    "RUNAWAY",
-    "CRASH",
-    "THROTTLE",
-    "ANGLE",
-    "BOOTGRACE",
-    "NOPREARM",
-    "LOAD",
-    "CALIB",
-    "CLI",
-    "CMS",
-    "BST",
-    "MSP",
-    "PARALYZE",
-    "GPS",
-    "RESCUE_SW",
-    "DSHOT_TELEM",
-    "REBOOT_REQD",
-    "DSHOT_BBANG",
-    "NO_ACC_CAL",
-    "MOTOR_PROTO",
-    "ARMSWITCH",
+    [LOG2(ARMING_DISABLED_NO_GYRO)] = "NOGYRO",
+    [LOG2(ARMING_DISABLED_FAILSAFE)] = "FAILSAFE",
+    [LOG2(ARMING_DISABLED_RX_FAILSAFE)] = "RXLOSS",
+    [LOG2(ARMING_DISABLED_NOT_DISARMED)] = "NOT_DISARMED",
+    [LOG2(ARMING_DISABLED_BOXFAILSAFE)] = "BOXFAILSAFE",
+    [LOG2(ARMING_DISABLED_RUNAWAY_TAKEOFF)] = "RUNAWAY",
+    [LOG2(ARMING_DISABLED_CRASH_DETECTED)] = "CRASH",
+    [LOG2(ARMING_DISABLED_THROTTLE)] = "THROTTLE",
+    [LOG2(ARMING_DISABLED_ANGLE)] = "ANGLE",
+    [LOG2(ARMING_DISABLED_BOOT_GRACE_TIME)] = "BOOTGRACE",
+    [LOG2(ARMING_DISABLED_NOPREARM)] = "NOPREARM",
+    [LOG2(ARMING_DISABLED_LOAD)] = "LOAD",
+    [LOG2(ARMING_DISABLED_CALIBRATING)] = "CALIB",
+    [LOG2(ARMING_DISABLED_CLI)] = "CLI",
+    [LOG2(ARMING_DISABLED_CMS_MENU)] = "CMS",
+    [LOG2(ARMING_DISABLED_BST)] = "BST",
+    [LOG2(ARMING_DISABLED_MSP)] = "MSP",
+    [LOG2(ARMING_DISABLED_PARALYZE)] = "PARALYZE",
+    [LOG2(ARMING_DISABLED_GPS)] = "GPS",
+    [LOG2(ARMING_DISABLED_RESC)] = "RESCUE_SW",
+    [LOG2(ARMING_DISABLED_DSHOT_TELEM)] = "DSHOT_TELEM",
+    [LOG2(ARMING_DISABLED_REBOOT_REQUIRED)] = "REBOOT_REQD",
+    [LOG2(ARMING_DISABLED_DSHOT_BITBANG)] = "DSHOT_BBANG",
+    [LOG2(ARMING_DISABLED_ACC_CALIBRATION)] = "NO_ACC_CAL",
+    [LOG2(ARMING_DISABLED_MOTOR_PROTOCOL)] = "MOTOR_PROTO",
+    [LOG2(ARMING_DISABLED_CRASHFLIP)] = "CRASHFLIP",
+    [LOG2(ARMING_DISABLED_ARM_SWITCH)] = "ARMSWITCH",
 };
 
+STATIC_ASSERT(ARRAYLEN(armingDisableFlagNames) == ARMING_DISABLE_FLAGS_COUNT, "armingDisableFlagNames size mismatch");
```

Comment:
Actually, using both designated initializers and STATIC_ASSERT won't work well.  in most cases `[LOG2(ARMING_DISABLED_ARM_SWITCH)] = ` will set array size to ARMING_DISABLE_FLAGS_COUNT, so STATIC_ASSERT will succeed. There may be hole ( = `NULL` in armingDisableFlagNames), it can be handled where necessary. This will only trigger in  case when `ARMING_DISABLED_<something> > ARMING_DISABLED_ARM_SWITCH`

So either:
- use designated initializers. It will prevent collision and invalid ordering of strings, but can leave empty/NULL names.
- check  array len - it will catch missing entry (which is common mistake after adding new enum value), but will ignore wrong order.

Reply:
Choose second option here :+1: